src/app/pages/swap/swap-container.tsx

```diff
@@ -1,66 +1,128 @@
-import { useState } from 'react';
+import { useCallback, useEffect, useMemo, useState } from 'react';
 import { Outlet, useNavigate } from 'react-router-dom';
 
-import BtcIcon from '@assets/images/btc-icon.png';
-import XBtcIcon from '@assets/images/xbtc-icon.png';
+import { AlexSDK, Currency, TokenInfo } from 'alex-sdk';
 import BigNumber from 'bignumber.js';
 
 import { createMoney } from '@shared/models/money.model';
 import { RouteUrls } from '@shared/route-urls';
 
+import { useAllTransferableCryptoAssetBalances } from '@app/common/hooks/use-transferable-asset-balances.hooks';
 import { whenPageMode } from '@app/common/utils';
-import { useNativeSegwitBalance } from '@app/query/bitcoin/balance/bitcoin-balances.query';
-import { useCurrentAccountNativeSegwitIndexZeroSigner } from '@app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks';
 
 import { SwapContainerLayout } from './components/swap-container.layout';
 import { SwapForm } from './components/swap-form';
 import { SwapAsset, SwapFormValues } from './hooks/use-swap';
-import { SwapContext, SwapProvider } from './swap.context';
-
-// TODO: Remove and set to initial state to 0 with live data
-const tempExchangeRate = 0.5;
```

Comment:
The exchange rate is not a constant, we have to call readonly function for each amount that user input.

We removed this field and added a function to fetch the toAmount.

Reply:
Yep, figured this would be removed, thx.